core/trino-main/src/main/java/io/trino/operator/exchange/UniformPartitionRebalancer.java

```diff
@@ -280,7 +280,7 @@ private boolean rebalancePartition(WriterId from, WriterId to)
                     if (partitionInfo.getWriterCount() <= numberOfWriters && estimatedPartitionWrittenBytes >= writerMinSize) {
```

Comment:
nit: we probably should update estimated written size for all writers that were writing to a given partition as you did in https://github.com/trinodb/trino/pull/15892/files#diff-075c4ea1c37a95efc0ba3c0e3a4472d83e92a0c6034257edc76e1ee9372520dc.

But I think we can merge without that improvement too